system-int-test/src/cmr/system_int_test/utils/variable_util.clj

```diff
@@ -0,0 +1,328 @@
+                               :update)))
+
+(defn- grant-permitted?
+  "Check if a given grant id is in the list of provided ACLs."
+  [grant-id acls]
+  (contains?
+    (into
+      #{}
+      (map :guid acls))
+    grant-id))
+
+(defn- group-permitted?
+  "Check if a given group id is in the list of provided ACLs."
+  [group-id acls]
+  (contains?
+    (reduce
+      #(into %1 (map :group-guid %2))
+      #{}
+      (map :aces acls))
+    group-id))
+
+(defn permitted?
+  "Check if a the ACLs for the given token include the given grant and group IDs."
+  [token grant-id group-id]
+  (let [acls (get-system-ingest-update-acls token)]
+    (and (grant-permitted? grant-id acls)
+         (group-permitted? group-id acls))))
+
+(defn not-permitted?
+  [& args]
+  (not (apply permitted? args)))
+
+(defn grant-all-variable-fixture
+  "A test fixture that grants all users the ability to create and modify variables"
+  [f]
+  (e/grant-all-variable (s/context))
+  (f))
+
+(def sample-variable
+  {:Name "A name"
+   :LongName "A long UMM-Var name"
+   :Units "m"
+   :DataType "float32"
+   :DimensionsName ["H2OFunc"
+                    "H2OPressureLay"
+                    "MWHingeSurf"
+                    "Cloud"]
+   :Dimensions ["11" "14" "7" "2"]
+   :ValidRange []
+   :Scale 1.0
+   :Offset 0.0
+   :FillValue -9999.0
+   :VariableType "SCIENCE_VARIABLE"})
+
+(defn make-variable
+  "Makes a valid variable based on the given input"
+  ([]
+   (make-variable nil))
+  ([attributes]
+   (merge sample-variable attributes))
+  ([index attribs]
+   (merge
+    sample-variable
+    {:Name (str "Name" index)
+     :Version (str "V" index)
+     :LongName (str "Long UMM-Var name " index)}
+    attribs)))
+
+(defn- assert-convert-kebab-case
+  "Assert that the field names in the map does not have dashes, and convert the given concept map
+  to use kebab case keys."
+  [concept-map]
+  ;; This is to assert that the variables api response will use underscore, not dash
+  (is (empty? (select-keys concept-map [:concept-id :revision-id :variable-name
+                                        :originator-id :variable-association :associated-item])))
+  (util/map-keys->kebab-case concept-map))
+
+(defn- kebab-case-body
+  "Returns the body with variables converted to kebab case."
+  [body]
+  (cond
+    (sequential? body) (map assert-convert-kebab-case body)
+    (map? body) (assert-convert-kebab-case body)
+    :else body))
+
+(defn- process-response
+  [{:keys [status body]}]
+  (let [body (kebab-case-body body)]
+    (if (map? body)
+      (assoc body :status status)
+      {:status status
+       :body body})))
+
+(defn create-variable
+  "Creates a variable."
+  ([token variable]
+   (create-variable token variable nil))
+  ([token variable options]
+   (let [options (merge {:raw? true :token token} options)]
+     (process-response (transmit-variable/create-variable (s/context) variable options)))))
+
+(defn update-variable
```

Comment:
Awesome -- had done something like this in my current branch; will merge with master when this PR gets approved and use this instead.